### PR TITLE
Client Simplifications And Cleanups

### PR DESCRIPTION
**Pull request by Bastian-Krause**
**Description**
Various simplifications and cleanups to make the client code smaller.

No change in client behavior intended.

**Checklist**
- [ ] Documentation for the feature
- [ ] CHANGES.rst has been updated
- [x] PR has been tested

### CODE REVIEW

**Comment by Bastian-Krause** [comment]
Force-pushed:
- add a test case for `Target.get_driver_or_new()`

**Comment by Bastian-Krause** [comment]
Rebased on current master.

**Comment by Bastian-Krause** [comment]
Rebased on current master.

**Comment by Bastian-Krause** [comment]
Rebased on current master.

**Review by Emantor — DISMISSED** [review]
Looks good to me.

**Comment by Bastian-Krause** [comment]
Ping?

File: labgrid/target.py [hunk 1]
```diff
@@ -219,6 +219,30 @@ def get_driver(self, cls, *, name=None, activate=True):
         """
         return self._get_driver(cls, name=name, activate=activate)
 
+    def get_driver_or_new(self, cls, *, name=None, activate=True, binding_map=None):
```
Comment (jluebbe) [inline labgrid/target.py:222]: As this is a special case for the client, I'd prefer to have this in the client module. For other use-cases, you either build a Target with Drivers in code or instantiate it from an Environment, but not a mix of both.

**Review by jluebbe — CHANGES_REQUESTED** [review]

File: labgrid/target.py (hunk 1 above)
Comment (Bastian-Krause) [reply labgrid/target.py]: Okay, I've moved this to the ClientSession.

**Review by jluebbe — APPROVED** [review]

**Review by Emantor — APPROVED** [review]

**Pull request merged**